src/server.py

```diff
@@ -182,22 +182,37 @@ def account_post():
         agoraModel.changeEmail(g.sessionToken, data['email'])
     return redirect("/account")
 
-@app.route('/backup/<code>', methods=['POST'])
-def backup_post(code):
+@app.route('/backup')
+def backup_get():
+    return render_template('recover-account.html', data=g.data)
+
+@app.route('/backup', methods=['POST'])
+def backup_post():
     data = request.form
-    if "email" in data:
-        agoraModel.backupRecover(code, data['email'])
+    if 'email' in data and 'code' in data:
+        agoraModel.backupRecover(data['code'], data['email'])
     return render_template('info.html', data=g.data, msg='backup-sent-email')
 
+@app.route('/changepass')
+def change_password_request_get():
+    g.data['initial_request'] = True
+    return render_template('reset-password.html', data=g.data)
+
 @app.route('/changepass', methods=['POST'])
-def change_password_request():
+def change_password_request_post():
     data = request.form
     if "email" in data:
         agoraModel.recoverAccount(data["email"])
     return render_template('info.html', data=g.data, msg='recovery-sent-email')
 
+@app.route('/changepass/<token>')
+def change_password_get(token):
```

Comment:
Instead of using the `initial_request` and `secondary_request`  keys of `data` to distinguish between different visits to `/changepass`, why not just use two different templates? The two forms look different enough, so I see no reason to have a single template with a case split rather than 2 different templates.

Reply:
yep I was thinking that as well. I think that's cleaner too.